### DISCUSSION TITLE
Connect events are not emitted?

### DISCUSSION THREAD

**Discussion by monsonjeremy** [wevm/wagmi#300] [Ideas]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

0.2.21

### Current Behavior

When a user connects a wallet using either the injected providers (wallet link or metamask) or the wallet connect provider, there is no `emit('connect')` called. This is useful for syncing with logic that may exist outside of react or need to be invoked using the event listeners (since `.on('connect')` is provided)

### Expected Behavior

Whenever a wallet is connected, there should be a call to `wallet.emit('connect')` in order to send those events.

### Steps To Reproduce

- go to the provided code-sandbox and connect a metamask or coinbase wallet
- see that the connect event is never emitted (on manual or auto connect)

### Link to Minimal Reproducible Example (CodeSandbox, StackBlitz, etc.)

https://codesandbox.io/s/holy-thunder-t4yzxb?file=/src/App.tsx

**Comment by tmm** [comment]
wagmi connectors aren’t [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) providers so aren’t required to emit events like this. That said the new version of wagmi will allow for subscribing to events in Vanilla JS outside of React.
